generator/src/main/java/com/scottlogic/deg/generator/walker/reductive/ReductiveDecisionTreeReducer.java

```diff
@@ -135,10 +136,18 @@ private boolean fixedValueConflictsWithAtomicConstraint(FixedField fixedField, A
         }
 
         FieldSpec mergedFieldSpec = merged.get();
+
+        if (mergedFieldSpec.getSetRestrictions() == null) {
+            return false;
+        }
+
         Set<Object> mergedFieldSpecWhitelist = mergedFieldSpec.getSetRestrictions().getWhitelist();
 
-        if (!mergedFieldSpecWhitelist.contains(fixedField.getCurrentValue())){
-            return true; //the fixedField value has been removed from the whitelist, the value contradicts another constraint
+        if (mergedFieldSpecWhitelist == null || !mergedFieldSpecWhitelist.contains(fixedField.getCurrentValue())) {
+            if(mergedFieldSpec.getNullRestrictions()!=null && mergedFieldSpec.getNullRestrictions().nullness != Nullness.MUST_NOT_BE_NULL){
+                return false; // the fixedField is not in the whitelist but can be null
```

Comment:
Can you simplify this by not using a double negative (i.e. assert that null-restrictions allow null, rather than not allowing not null)

Reply:
i.e. can this be:

```suggestion
if (mergedFieldSpec.getNullRestrictions() != null && mergedFieldSpec.getNullRestrictions().nullness == Nullness.MUST_BE_NULL){
                return false; // the fixedField has a value, but must be null
```

Reply:
sure, i also added a check for if the null restriction is itself null in which case, we don't have a null restriction so it can still be null and should be omitted.